Documentation/sourcebuild-in-repos/new-dependencies.md

```diff
@@ -0,0 +1,128 @@
+intermediate package and the CoherentParentDependency attribute should be
+set to the repo that consumes the dependency. For an example,
+see [installer's F# dependency](https://github.com/dotnet/installer/blob/ba1739a2363b1062f03ea386ec67174c6468d3b2/eng/Version.Details.xml#L128).
+You can find the version needed by running `darc get-build`
+or using [BAR](https://aka.ms/bar).
+
+## Microsoft dependencies
+
+.NET dependencies are from a Microsoft .NET project, like
+[dotnet/roslyn](https://github.com/dotnet/roslyn),
+[dotnet/runtime](https://github.com/dotnet/runtime),
+and [dotnet/fsharp](https://github.com/dotnet/fsharp).  These repos use
+Arcade to build and publish.  Adding a new .NET dependency is usually
+quite easy.  Internal dependencies are from Microsoft repos but not .NET,
+like [Microsoft/msbuild](https://github.com/dotnet/msbuild),
+[NuGet/NuGet.Client](https://github.com/NuGet/NuGet.Client),
+and [Microsoft/ApplicationInsights](https://github.com/Microsoft/ApplicationInsights-dotnet).
+Adding a new internal dependency is usually pretty straightforward but does
+have some edge cases.  Use this checklist:
+
+1. Are you already using a package from the same repo and is the new
+  dependency already source-built?  You can determine this by checking if
+  the repo has a SourceBuild leg building, e.g. runtime's is
+  `runtime-dev-innerloop (Build Linux x64 release SourceBuild)`.
+
+    1. This is the simplest case.  You can add new dependencies from repos
+      you are already using freely.  Note that you need to use the same version
+      for all packages from the repo.
+1. Is the repo already built in source-build including the specific
+  package you want?
+    1. Add the dependency using `darc add-dependency` as normal, then
+      add the [source-build metadata](#Basics) as above.
+1. Is the repo already built in source-build but the specific package is not?
+    1. There's probably an issue with source-building this package.  Please
+      talk to a [source-build team member](https://github.com/orgs/dotnet/teams/source-build-internal)
+      about why that is and whether we can fix it.
+1. Is this a Microsoft repo that uses Arcade to build?
+    1. Does the foreign repo depend on your repo, directly or indirectly?
+      i.e. would adding the dependency create a cycle?
+        1. This isn't necessarily a deal-breaker - it can sometimes be worked
+          around with reference-only packages.  Please contact a
+          [source-build team member](https://github.com/orgs/dotnet/teams/source-build-internal)
+          to discuss.
+    1. Does the foreign repo publish to BAR?
+        1. If not, please contact them to get them publishing to BAR
+          in an appropriate channel.
+    1. If neither of these caveats apply you should be in good shape.
+      Follow the instructions under "Basics" above.
+1. Source-build has in the past used Arcade shims to allow non-Arcade repos
```

Comment:
If we combine the sections, I think text only packages should be inserted here.  The last case should discuss the option to use an Arcade shim or use source-build externals.  Either case, logging an issue to begin the discussion is the action to take.

Reply:
Changed as suggested.